### PR TITLE
MOTECH-2163 Add metrics module

### PR DESCRIPTION
**Pull request by justin-hayes**
This module provides an integration for the Dropwizard
metrics library, exposes an API to register metrics with
the system, and allows implementers to stream metrics to Graphite.

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by justin-hayes** [comment]
This pull request is the result of moving a previous PR to the core platform (https://github.com/motech/motech/pull/331) to the modules repository. This submission has also been updated to account for code review at the link above.

**Comment by pgesek** [comment]
test this please

File: motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Test
+    public void shouldRegisterDefaultHealthCheck() {
+        Set<String> registered = healthCheckRegistryService.getNames();
+
+        assertNotNull(registered);
+
+        System.out.println(registered);
+
+        assertTrue(registered.size() == 1);
```
Comment (sebbrudzinski) [inline motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java]: Use assertEquals

File: motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            @Override
+            public Result check() throws HealthCheckException {
+                return null;
+            }
+        });
+
+        Set<String> registered = healthCheckRegistryService.getNames();
+
+        assertNotNull(registered);
+        assertTrue(registered.size() == 2);
```
Comment (sebbrudzinski) [inline motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java]: Use assertEquals

File: motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        assertTrue(registered.contains("foo"));
+
+        healthCheckRegistryService.unregister("foo");
+
+        Set<String> unregistered = healthCheckRegistryService.getNames();
+
+        assertTrue(unregistered.size() == 1);
```
Comment (sebbrudzinski) [inline motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java]: Use assertEquals

File: motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        assertNotNull(healthCheckRegistryService);
+    }
+
+    @Test
+    public void shouldRegisterDefaultHealthCheck() {
+        Set<String> registered = healthCheckRegistryService.getNames();
+
+        assertNotNull(registered);
+
+        System.out.println(registered);
```
Comment (sebbrudzinski) [inline motech-metrics/src/test/java/org.motechproject.metrics/it/HealthCheckRegistryServiceBundleIT.java]: Do not use printing to standard output and use loggers instead.
You can extend BasePaxIT to use getLogger() from there.

**Comment by pgesek** [comment]
test this please

**Comment by pgesek** [comment]
needs merge with master

**Comment by llavoie** [comment]
@justin-hayes are you able to finish ushering this through or do you need someone to take it over?

**Comment by sebbrudzinski** [comment]
test this please

**Comment by sebbrudzinski** [comment]
@pgesek Looks OK to me codewise, but we might want to wait with merge until we decide what to do with the other metrics module

**Comment by pgesek** [comment]
I'm going to merge this so that @justin-hayes  won't have to keep updating the branch. The merging of these two modules is not important enough to hold this off, we can merge the modules afterwards.

**Pull request merged**